### ISSUE TITLE
More annotation colors

### ISSUE THREAD

**Issue by danfishgold** [CodeReadingClubs/annotation-tool#3]
Currently there are only four colors for markers/arrows and only three for line annotations (i.e. gutter annotations).

These are not enough and they aren't super accessible for colorblind people.

This would also be a good opportunity to allow customization of the amount of line annotation options, which would be useful for facilitators in the "most important lines" exercise.

**Comment by danfishgold** [comment]
Turns out this won't be a good opportunity to customize the amount of line annotation options, because the line-annotations div is a part of the svg, and so if it changes size then all the markers and arrows shift. In order to implement this we would need to make it so that the svg only covers the `<pre>` elements, which is easy enough, but it would also be a breaking change (although we can hotfix it by shifting all markers and arrows a bit to the right. This would work but only for people who use the default zoom setting.)

In summary: I think we should do it, but it would be a substantial change so it should be separated from "more annotations colors"

**Comment by danfishgold** [comment]
![Screen Shot 2021-08-10 at 14 39 56](https://user-images.githubusercontent.com/777023/128860712-571c291a-6de0-4d4d-ac95-0e38c43f1566.png)

@katjam what do you think?
I used colors from the [Material Design color palettes](https://material.io/design/color/the-color-system.html#tools-for-picking-colors)

**Comment by katjam** [comment]
I like the palette, but I wonder it the light and dark of each have enough meaningful / visible difference?

I thought for example it could be nice to colour the instantiation a dark red and all uses of it in light read, bit those tints are too close for that.

If it's an aesthetic choice to use one or the other (top or bottom row) in terms of general readability for an individual, then having a top level toggle of 'palette' would be better than presenting all on them in the modal.

**Comment by danfishgold** [comment]
![Screen Shot 2021-08-15 at 22 31 17](https://user-images.githubusercontent.com/777023/129490194-4fe798f5-c7de-4f63-b195-c0379065bc5c.png)

@katjam what about this? it's 16 different hues with the same brightness (that you could configure in a top level toggle, like you said) (this would also work nicely with a dark theme toggle, where we could automatically turn the 100 level (in the material color palettes) to a 800 and the 200 to a 700 for example)

**Comment by katjam** [comment]
Sounds good to me.
